runtime/hydradx/src/evm/erc20_currency.rs

```diff
@@ -259,15 +274,35 @@ where
 		amount: Self::Balance,
 	) -> sp_runtime::DispatchResult {
 		let sender = <pallet_evm_accounts::Pallet<T>>::evm_address(from);
-		<Self as ERC20>::transfer(
-			CallContext {
-				contract,
-				sender,
-				origin: sender,
-			},
-			EvmAccounts::<T>::evm_address(to),
-			amount,
-		)
+		let call_context = CallContext {
+			contract,
+			sender,
+			origin: sender,
+		};
+
+		if let Some(underlying_asset) = AaveTradeExecutor::<T>::get_underlying_asset_from(contract) {
+			let asset_id = HydraErc20Mapping::address_to_asset(contract)
```

Comment:
move all this stuff in this branch to aave trade executor new "transfer" function